app/services/export/CalendarExporter.java

```diff
@@ -0,0 +1,153 @@
+  private static final String DATE_STAMP = "DTSTAMP:";  // TODO format: 20060812T125900Z
+
+  private static final Map<String, String> mFieldMap = new HashMap<>();
+  static{
+    mFieldMap.put(UID, JsonLdConstants.ID);
+    mFieldMap.put(COMMON_NAME, "organizer.name.@value");
+    mFieldMap.put(SUMMARY, "name.@value");
+    mFieldMap.put(DESCRIPTION, "description.@value");
+    mFieldMap.put(URL, "url");
+    mFieldMap.put(DATE_START, "startDate");
+    mFieldMap.put(DATE_END, "endDate");
+    mFieldMap.put(LOCATION, "location.address.streetAddress".concat(VALUE_SEPARATOR)
+                    .concat("location.address.postalCode").concat(VALUE_SEPARATOR)
+                    .concat("location.address.addressLocality").concat(VALUE_SEPARATOR)
+                    .concat("location.address.addressRegion").concat(VALUE_SEPARATOR)
+                    .concat("location.address.addressCountry").concat(VALUE_SEPARATOR));
+
+    mFieldMap.put(GEO, "location.geo.lat".concat(VALUE_SEPARATOR).concat("location.geo.lon"));
+  }
+
+  private final Locale mPreferredLocale;
+
+  public CalendarExporter(Locale aPreferredLocale){
+    mPreferredLocale = aPreferredLocale;
+  }
+
+  @Override
+  public String export(Resource aResource) {
+    if (!aResource.getType().equals("Event")) {
+      return null;
+    }
+    StringBuilder result = new StringBuilder(HEADER);
+    result.append(exportResourceWithoutHeader(aResource));
+    result.append(FOOTER);
+    return result.toString();
+  }
+
+  @Override
+  public String export(ResourceList aResourceList) {
+    StringBuilder result = new StringBuilder(HEADER);
+    for (Resource resource : aResourceList.getItems()){
+      if (resource.getType().equals("Event")){
+        result.append(exportResourceWithoutHeader(resource));
+      }
+    }
+    result.append(FOOTER);
+    return result.toString();
+  }
+
+  private String exportResourceWithoutHeader(Resource aResource){
+    StringBuilder result = new StringBuilder(EVENT_BEGIN);
+    for (Map.Entry<String, String> mapping : mFieldMap.entrySet()){
+      boolean hasAppendedSomething = false;
+      String[] mappingValues = mapping.getValue().split(VALUE_SEPARATOR);
+      for (int i=0; i<mappingValues.length; i++){
+        final Object o = aResource.getNestedFieldValue(mappingValues[i], mPreferredLocale);
+        if (o != null) {
+          String value = o.toString();
+          if (i == 0){
+            result.append(mapping.getKey());
+          } //
+          else{
+            result.append(VALUE_SEPARATOR);
+          }
+          result.append(value);
+          hasAppendedSomething = true;
+        }
+      }
+      if (hasAppendedSomething){
+        result.append("\n");
+      }
+    }
+    result.append(getExportedOrganizer(aResource));
+    result.append(CAL_CLASS);
+    result.append(DATE_STAMP).append(System.currentTimeMillis()).append("\n"); // TODO change time stamp format
```

Comment:
We should probably not use the system time here, but the time the resource was last modified. Currently we don't have this easily accessible though at this place so we might simply omit it for now. EDIT: DTSTAMP is required, so we have to get something sensible in there.